supabase/functions/_backend/private/sso.ts

```diff
@@ -0,0 +1,539 @@
+  metadata_url: z.optional(z.nullable(z.string())),
+  enabled: z.optional(z.boolean()),
+})
+
+export const app = new Hono<MiddlewareKeyVariables>()
+
+app.use('/*', useCors)
+
+// Helper function to check Enterprise plan
+async function checkEnterprisePlan(c: any, orgId: string): Promise<boolean> {
+  const planName = await getCurrentPlanNameOrg(c, orgId)
+  return planName === 'Enterprise'
+}
+
+// ============================================================================
+// SSO PROVIDER ENDPOINTS
+// ============================================================================
+
+// GET /sso/config - Get SSO config for an org
+app.get('/config', middlewareV2(['all', 'write', 'read']), async (c) => {
+  const auth = c.get('auth')!
+  const orgId = c.req.query('org_id')
+
+  if (!orgId) {
+    return simpleError('missing_org_id', 'org_id query parameter is required')
+  }
+
+  const supabaseAdmin = await useSupabaseAdmin(c)
+
+  // Check user has at least admin rights
+  const userRight = await supabaseAdmin.rpc('check_min_rights', {
+    min_right: 'admin',
+    org_id: orgId,
+    user_id: auth.userId,
+    channel_id: null as any,
+    app_id: null as any,
+  })
+
+  if (userRight.error || !userRight.data) {
+    return quickError(401, 'not_authorized', 'Not authorized to view SSO config')
+  }
+
+  // Get SSO config
+  const { data, error } = await supabaseAdmin
+    .from('org_sso_providers')
+    .select('*')
+    .eq('org_id', orgId)
+    .maybeSingle()
+
+  if (error) {
+    return simpleError('get_sso_config_error', 'Failed to get SSO config', { error })
+  }
+
+  // Check if org is Enterprise
+  const isEnterprise = await checkEnterprisePlan(c, orgId)
+
+  return c.json({
+    config: data,
+    is_enterprise: isEnterprise,
+  })
+})
+
+// POST /sso/config - Create or update SSO config
+app.post('/config', middlewareV2(['all', 'write']), async (c) => {
+  const auth = c.get('auth')!
+  const body = await parseBody<any>(c)
+
+  const parsedBody = ssoProviderSchema.safeParse(body)
+  if (!parsedBody.success) {
+    return simpleError('invalid_body', 'Invalid request body', { errors: parsedBody.error })
+  }
+
+  const { org_id, ...ssoConfig } = parsedBody.data
+  const supabaseAdmin = await useSupabaseAdmin(c)
+
+  // Check user has super_admin rights
+  const userRight = await supabaseAdmin.rpc('check_min_rights', {
+    min_right: 'super_admin',
+    org_id,
+    user_id: auth.userId,
+    channel_id: null as any,
+    app_id: null as any,
+  })
+
+  if (userRight.error || !userRight.data) {
+    return quickError(401, 'not_authorized', 'Super admin rights required')
+  }
+
+  // Check Enterprise plan
+  if (!await checkEnterprisePlan(c, org_id)) {
+    return quickError(403, 'requires_enterprise', 'SSO requires Enterprise plan')
+  }
+
+  // Upsert SSO config
+  const { data, error } = await supabaseAdmin.rpc('upsert_org_sso_provider', {
+    p_org_id: org_id,
+    p_supabase_sso_provider_id: ssoConfig.supabase_sso_provider_id,
```

Comment:
We can add a `p_user_id`
```suggestion
  const { data, error } = await supabaseAdmin.rpc('upsert_org_sso_provider', {
    p_user_id: auth.userId,
    p_org_id: org_id,
    p_supabase_sso_provider_id: ssoConfig.supabase_sso_provider_id,
```